### PR TITLE
Different validator

### LINKED ISSUE #1: Decide on a validator
Not too sure about jsen at the moment.

### LINKED ISSUE #3: Add a custom format for versions
The regex for versions takes us only so far. Rather than trying to match ranges with a regex it seems like it would be saner to use a custom format instead.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #1 
Fixes #3

### CODE REVIEW

File: src/validator.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import fs from 'fs';
+import ajv from 'ajv';
+import { isValidVersionString } from './formats';
+
+export var schema = fs.readFileSync(
+  path.join(__dirname, 'manifest-schema.json'), 'utf8');
+
+var validator = ajv({allErrors: true});
+validator.addFormat('versionString', isValidVersionString);
+export default validator.compile(JSON.parse(schema));
```
Comment (tofumatt) [inline src/validator.js]: This would be easier to read with a newline above it, I think.

File: tests/test.formats.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  it('should be valid version string', () => {
+    assert.ok(isValidVersionString('1'));
+    assert.ok(isValidVersionString('1.0'));
+    assert.ok(isValidVersionString('2.10.2'));
+    assert.ok(isValidVersionString('3.1.2.4567'));
+    assert.ok(isValidVersionString('3.1.2.65535'));
+  });
+
+  it('should not be valid version string', () => {
```
Comment (tofumatt) [inline tests/test.formats.js]: I'd add a test for things that are valid semver (stuff like `1.0.0-beta2` or `2.0.0-rc`, which are valid as I understand it) so it's clear that we aren't exactly emulating semver.

**Comment by tofumatt** [comment]
r+wc

File: tests/test.formats.js (hunk 2 above)
Comment (muffinresearch) [inline tests/test.formats.js]: Yep good call.

**Pull request merged**